src/Illuminate/Http/Resources/CollectsResources.php

```diff
@@ -52,8 +52,8 @@ protected function collects()
         if ($this->collects) {
             $collects = $this->collects;
         } elseif (str_ends_with(class_basename($this), 'Collection') &&
-            (class_exists($class = Str::replaceLast('Collection', '', get_class($this))) ||
-             class_exists($class = Str::replaceLast('Collection', 'Resource', get_class($this))))) {
+            (@class_exists($class = Str::replaceLast('Collection', '', get_class($this))) ||
+             @class_exists($class = Str::replaceLast('Collection', 'Resource', get_class($this))))) {
```

Comment:
We should not suppress errors like this.

Reply:
without supression this throws an exception and never reaches the second replacement (Collection → Resource)

Reply:
@taylorotwell class_exists() throws an error here, because its called with autoload.. and that fails if the class name is not a real name for a class. When you wrote this, you forgot to test for the second case ;)

this only works for `..\Resource\SomeCollection` using `..\Resource\Some`
and will fail when I am using `..\Resource\SomeResource`

Reply:
@driesvints without suppression this throws an exception and never reaches the second replacement (Collection → Resource)